src/main/java/Duke.java

```diff
@@ -1,10 +1,107 @@
+import java.util.Scanner;
+
 public class Duke {
+
+    private static final int TODO_DONE_LENGTH = 5;
+    private static final int DEADLINE_LENGTH = 9;
+    private static final int EVENT_LENGTH = 6;
+    private static final int BY_AT_LENGTH = 3;
+    private static Task[] tasksList = new Task[100];
+    private static int taskCounter = 0;
+
     public static void main(String[] args) {
+        showWelcomeMessage();
+        inputLoop();
+    }
+
+    private static void inputLoop() {
+        Scanner in = new Scanner(System.in);
+        String line;
+        while (true) {
+            line = in.nextLine();
+            if (line.equals("bye")) {
+                showExitMessage();
+                break;
+            } else if (line.equals("list")) {
+                printTasksList();
+                continue;
+            } else if (line.startsWith("done")) {
+                int index = Integer.parseInt(line.substring(TODO_DONE_LENGTH)) - 1;
+                tasksList[index].markAsDone();
+                printDoneTask(index);
+                continue;
+            } else if (line.startsWith("todo")) {
+                String description = line.substring(TODO_DONE_LENGTH);
+                Task newTask = new ToDo(description);
+                tasksList[taskCounter] = newTask;
+            } else if (line.startsWith("deadline")) {
+                int index = line.indexOf("/");
+                String description = line.substring(DEADLINE_LENGTH, index - 1);
+                String by = line.substring(index + BY_AT_LENGTH).trim();
+                Task newTask = new Deadline(description, by);
+                tasksList[taskCounter] = newTask;
+            } else if (line.startsWith("event")) {
+                int index = line.indexOf("/");
+                String description = line.substring(EVENT_LENGTH, index - 1);
+                String at = line.substring(index + BY_AT_LENGTH).trim();
+                Task newTask = new Event(description, at);
+                tasksList[taskCounter] = newTask;
+            }
+            taskCounter++;
+            printNewTask();
+        }
+    }
+
+    private static void printDoneTask(int index) {
+        printLine();
+        System.out.println("\tNice! I've marked this task as done:");
+        System.out.print("\t" + "  ");
```

Comment:
I don't think there is a need for concatenation here, as "\t  " should work just fine

Reply:
oh true thank you